src/components/TodoHeader.tsx

```diff
@@ -0,0 +1,86 @@
+import classNames from 'classnames';
+import React, { useCallback, useMemo, useState } from 'react';
+import { Todo } from '../types/Todo';
+import { ErrorMessage } from '../types/Error';
+import { USER_ID } from '../utils/USER_ID';
+
+type Props = {
+  todos: Todo[],
+  onUpdate: (todo: Todo) => Promise<void>,
+  onAdd: (todo: Omit<Todo, 'id'>) => Promise<void>,
+  onError: (error: ErrorMessage) => void;
+};
+
+export const TodoHeader: React.FC<Props> = ({
+  todos, onUpdate, onAdd, onError,
+}) => {
+  const [newTitle, setNewTitle] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+
+  const isSomeVisible = useMemo(() => !!todos.length, [todos]);
```

Comment:
here useMemo is not really necessary
it can be beneficial if the todos array is large, but i doubt that this situation is possible in the project)
no need to fix this, just fyi